페어리뷰용 PR

jinny-l opened the pull request.

jinny-l committed:
📝docs: 4단계 요구사항 및 기능 목록 작성

jinny-l committed:
🚚rename(리뷰반영): DTO 패키지명을 소문자로 수정

jinny-l committed:
🚚rename: 로그인 관련 html 파일을 templates 폴더로 이동

* 참고: 폴더 이동 후 html 공통 레이아웃 적용

jinny-l committed:
📝docs: 새로운 기능의 URL 컨벤션 추가 및 기존 URL 컨벤션 수정

jinny-l committed:
♻️refactor(리뷰반영): 변경된 URL 컨벤션에 따라 매핑 URL 수정

jinny-l committed:
✨feat: 로그인 페이지 URL 매핑

jinny-l committed:
✨feat: 로그인 기능 구현

jinny-l committed:
✨feat: 로그인 상태에 따라 네비게이션 바를 다르게 출력하는 기능 구현

* 상세 기능:
  - 로그인 상태 시 상단 메뉴에서 “로그아웃”, “개인정보수정” 표시
  - 로그아웃 상태 시 “로그인”, “회원가입” 표시

jinny-l committed:
✨feat: 로그아웃 기능 구현

takealook97 commented on src/main/java/kr/codesqaud/cafe/controller/ArticleController.java:
```diff
@@ -18,12 +18,12 @@ public ArticleController(ArticleService articleService) {
-    @PostMapping("/article/write")
+    @PostMapping("/articles")
     public String write(final ArticleDTO articleDTO) {
         Article article = new Article(articleDTO.getWriter(), articleDTO.getTitle(), articleDTO.getContents());
```
posting을 할때 필요한 파라미터들을 담은 request dto를 따로 만드는 방법은 어떨까요? 현재는 세개이지만 메서드명이 길어지고 파라미터가 많아질수록 너무 길어질 것 같아서요!

takealook97 commented on src/main/java/kr/codesqaud/cafe/dto/LoginFormDTO.java:
```diff
@@ -0,0 +1,28 @@
+
+import javax.validation.constraints.Size;
+
+public class LoginFormDTO {
```
DTO라는 약어는 전부 대문자로 쓰는 것이 일반적인 관례이기에 "LoginFormDTO"와 같은 방식으로 쓰는 것도 가능하지만, 클래스명에서는 카멜표기법으로 "LoginFormDto"와 같이 대문자와 소문자를 혼용하는 방식으로 쓰는 것이 더 일반적이라고 합니다! - by chatgpt

takealook97 reviewed (commented):
추가로 이전 커밋이라 다루지 못한 내용이지만 optional 을 반환할때 get() 메서드로 바로 안에 있는 값을 꺼내어 사용하는 것을 지양할 필요가 있다고 합니다. 만약 값이 존재하지 않는데 get() 메서드를 호출한다면 NoSuchElementException이 발생하기에 Optional의 orElse()나 orElseGet() 메서드를 사용하여 값이 존재하지 않는 경우 기본 값을 설정할 수 있습니다!

잘봤습니다~~ 🙌

jinny-l committed:
🔧chore: 테스트 환경에서 DB을 실행하기 위한 환경 세팅

jinny-l committed:
✅test: LoginService 테스트 코드 작성

* test case:
  - 가입된 ID로 로그인할 때 비밀번호가 맞다면 로그인이 성공한다.
  - 가입된 ID로 로그인할 때 비밀번호가 틀리다면 로그인이 실패한다.
  - 가입되지 않은 ID로 로그인하면 로그인이 실패한다.

jinny-l committed:
✅test: UserService 테스트 코드 작성

* test case:
  - 회원 가입을 하면 회원 정보가 저장소에 저장된다.
  - 중복 ID로 회원 가입하면 예외가 발생한다.

jinny-l committed:
✅test: UserRepositoryImpl 테스트 코드 작성

* test case:
  - 회원 가입을 하면 회원 정보가 저장된다.
  - 회원 ID로 가입 정보를 찾을 수 있다.

jinny-l committed:
📝docs: 4단계에서 구현한 내용 반영
